jmh-benchmarks/src/main/java/org/apache/kafka/jmh/acl/AclAuthorizerBenchmark.java

```diff
@@ -115,45 +122,62 @@ private void setFieldValue(Object obj, String fieldName, Object value) throws Ex
 
             Set<AclEntry> entries = aclEntries.computeIfAbsent(resource, k -> new HashSet<>());
 
-            for (int aclId = 0; aclId < aclCount; aclId++) {
-                AccessControlEntry ace = new AccessControlEntry(principal.toString() + aclId,
-                    "*", AclOperation.READ, AclPermissionType.ALLOW);
-                entries.add(new AclEntry(ace));
+            for (int aclId = 0; aclId < aclCount / 2; aclId++) {
+                String acePrinciple = principal.toString() + (aclId == 0 ? "" : aclId);
+                AccessControlEntry allowAce = new AccessControlEntry(
+                    acePrinciple,
+                    "*", AclOperation.WRITE, AclPermissionType.ALLOW);
+                AccessControlEntry denyAce = new AccessControlEntry(
+                    acePrinciple,
+                    "*", AclOperation.WRITE, AclPermissionType.DENY);
```

Comment:
We probably want to retain the old benchmark as-is and add a different one for `authorizeByResourceType`. We were testing a common pattern before, but now we seem to be testing a very unlikely scenario. While this may be useful for testing `authorizeByResourceType`, it is not what we want for regression testing the authorizer.

Reply:
I think adding some dominant denies won't change the performance pattern of AclAuthorizer::acls and AclAuthorizer::authorize. 
1. AclAuthorizer::acls just return all the matching acls by the filter rule.  The portion btw "allow" and "deny" resources doesn't matter.
2. AclAuthorizer::authorize will iterate the and filter out the allow and deny ACEs respectively. Since it's using ResourcePattern as its indexing method, the portion btw "allow" and "deny" resources doesn't matter as well.